### PR TITLE
replace blocked methods with a proxy object that yields itself, to av…

### PR DESCRIPTION
**Pull request by snyderp**
…oid breaking code that expects the feature to be in place

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by snyderp** [comment]
Possible fix for #10285

File: app/extensions/brave/content/scripts/blockCanvasFingerprinting.js [hunk 1]
```diff
@@ -73,6 +73,72 @@ if (chrome.contentSettings.canvasFingerprinting == 'block') {
...
+    if (hint === 'number') {
+      return 0
+    }
+    if (hint === 'string') {
+      return ''
+    }
+    return undefined
+  }
+
+  var allPurposeProxy
```
Comment (diracdeltas) [inline app/extensions/brave/content/scripts/blockCanvasFingerprinting.js]: this can be combined into the line below

**Comment by diracdeltas** [comment]
do you have any test pages for this? i'll review in more detail later. thanks!

**Comment by snyderp** [comment]
The `proxy fingerprinting method` test now checks for this

**Comment by diracdeltas** [comment]
The new proxy test seems to time out: https://travis-ci.org/brave/browser-laptop/jobs/260855561#L3407. It may be intermittent. I can help investigate later.

File: app/extensions/brave/content/scripts/blockCanvasFingerprinting.js [hunk 2]
```diff
@@ -73,6 +73,71 @@ if (chrome.contentSettings.canvasFingerprinting == 'block') {
...
+      return unconfigurablePropNames
+    },
+    has: function (target, property) {
+      return (unconfigurablePropNames.indexOf(property) > -1)
+    },
+    getOwnPropertyDescriptor: function (target, property) {
+      if (unconfigurablePropNames.indexOf(property) === -1) {
+        return undefined
+      }
+      return Object.getOwnPropertyDescriptor(defaultFunc, property);
```
Comment (diracdeltas) [inline app/extensions/brave/content/scripts/blockCanvasFingerprinting.js]: minor: extra semicolon

**Comment by psnyde2** [comment]
I made some very minor formatting changes (to the blocking extensions, and to the unit test).

I'm not seeing the unit test time out though.  When I run `npm test -- --grep="proxy fingerprinting method"` things seem to be a-ok.  I do see other intermittent test failures though, but they seem to be unrelated to these changes (ie I see the same failures w/o this change)

**Comment by luixxiul** [comment]
https://github.com/brave/browser-laptop/pull/10287#issuecomment-320794723

> The new proxy test seems to time out: https://travis-ci.org/brave/browser-laptop/jobs/260855561#L3407. It may be intermittent. I can help investigate later.

It seems that the it fails on Travis because on Travis WebGL does not run. Other fingerprinting protection tests have the same issue, so we don't have these const: https://github.com/luixxiul/browser-laptop/blob/797f28c76705f77e0e34c1349faf3fcb974e39c5/test/bravery-components/braveryPanelTest.js#L146-L157

See https://github.com/brave/browser-laptop/pull/11360 for a suggestion.

**Comment by diracdeltas** [comment]
lifting to 0.19.x since it is needed for https://github.com/brave/browser-laptop/pull/11784

**Comment by diracdeltas** [comment]
0.19.x: e790a956a01eca4448bcd9edfb32b160510583e5

**Pull request merged**